### DISCUSSION TITLE
brew install mongosh downloading hundreds of files

### DISCUSSION THREAD

**Discussion by JESii** [Homebrew/discussions#2582] [Everyday usage]
I did a `brew install mongosh`, brew updated as expected, then proceeded to install mongosh.

However, once it completed the mongosh installation, it's gone on to download literally hundreds of files -- never seen anything like this before. Here's a sample:
```
==> Downloading https://ghcr.io/v2/homebrew/core/nghttp2/manifests/1.46.0
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/nghttp2/blobs/sha256:da456fbcfd7be04c28fe0882f7ba187db2c584ccc3eabc5ab631ac8278513984
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:da456fbcfd7be04c28fe0882f7ba187db2c584ccc3eabc5ab631ac8278513984?se=2021-12-03T05%3A10%3A00Z&sig=EX8y35LU
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/node/manifests/17.0.1-1
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/node/blobs/sha256:5dd16b4fb13f717e03753a42302b96140c609d5f5e8c77ff7d5465b006eb7551
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:5dd16b4fb13f717e03753a42302b96140c609d5f5e8c77ff7d5465b006eb7551?se=2021-12-03T05%3A10%3A00Z&sig=aecvahvo
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/py3cairo/manifests/1.20.1
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/py3cairo/blobs/sha256:ea6e1887539c142f3b24890521e9181fac8738d5fa2344c9e4c0734ea5b2b9a8
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:ea6e1887539c142f3b24890521e9181fac8738d5fa2344c9e4c0734ea5b2b9a8?se=2021-12-03T05%3A10%3A00Z&sig=Nx%2BrjD
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/pygobject3/manifests/3.42.0_1
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/pygobject3/blobs/sha256:f9f1c60b23478993fc7ad3df87314c44dd172bcce440c4fccaf4222849834541
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:f9f1c60b23478993fc7ad3df87314c44dd172bcce440c4fccaf4222849834541?se=2021-12-03T05%3A10%3A00Z&sig=JJwhBx2u
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/redis/manifests/6.2.6
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/redis/blobs/sha256:246f73498993a2a0c6c4326a298d2fcc3da6d61904ad09a631aa9c63a6800f76
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:246f73498993a2a0c6c4326a298d2fcc3da6d61904ad09a631aa9c63a6800f76?se=2021-12-03T05%3A10%3A00Z&sig=1UTmV1mb
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/boost/manifests/1.76.0
```
Has anyone seen anything like this? Thanks!

**Comment by Bo98** [comment]
You probably have a extremely outdated set of formulae (see `brew outdated`) and installing mongosh probably downloaded a newer version of some dependencies. As a result, in order to make sure nothing becomes broken from these newer versions, `brew` will by default update other things that used that dependency, as this is sometimes required for some libraries (e.g. if there's a change that's not backwards binary compatible, which is quite common for some libraries).

Looking at the list of formulae you've posted, some haven't been updated on your system for at least several months. This can lead to a large list of affected formulae when using a rolling release package manager like `brew`.

**Comment by carlocab** [comment]
See https://docs.brew.sh/FAQ#why-does-brew-upgrade-formula-or-brew-install-formula-also-upgrade-a-bunch-of-other-stuff. If you don't like this behaviour, you can try using `HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK`. See [`man brew`](https://docs.brew.sh/Manpage) for details.

Note that using `HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK` can and does lead to broken installations after upgrading software you have installed. These can typically be fixed by doing `brew upgrade`, but the road to remembering to do that can be frustrating and painful.

A better solution would be to regularly do `brew update && brew upgrade` to avoid having lots of upgraded formulae when you actually do need to `brew install` something.
